.github/workflows/trigger_rebase.yml

```diff
@@ -0,0 +1,42 @@
+name: Trigger Rebase
+
+on:
+  push:
+    branches:
+      - develop
+env:
+  AUTO_REBASE_PERSONAL_ACCESS_TOKEN: ${{ secrets.AUTO_REBASE_PERSONAL_ACCESS_TOKEN }}
+jobs:
+  fetch_all_prs_awaiting_automerge:
+    runs-on: ubuntu-latest
+    outputs:
+      pr_numbers: ${{ steps.get-all-awaiting-pr-numbers.outputs.pr_numbers }}
+    steps:
+      - name: Get all awaiting PR numbers
+        id: get-all-awaiting-pr-numbers
+        run: |
+          PR_NUMBERS=$(curl https://api.github.com/repos/UKGovernmentBEIS/beis-report-official-development-assistance/pulls\?state\=open | jq -c '.[] | select(.auto_merge!=null) | .number | [.]')
+          echo "::set-output name=pr_numbers::$PR_NUMBERS"
+  trigger_rebase_action:
+    if: ${{ needs.fetch_all_prs_awaiting_automerge.outputs.pr_numbers != '' }}
+    strategy:
+      matrix:
+        pr_number: ${{ fromJson(needs.fetch_all_prs_awaiting_automerge.outputs.pr_numbers) }}
+    needs: fetch_all_prs_awaiting_automerge
+    runs-on: ubuntu-latest
+    steps:
+      - name: Get the latest ref for ${{ matrix.pr_number }}
+        run: |
+          PR_DATA=$(curl https://api.github.com/repos/UKGovernmentBEIS/beis-report-official-development-assistance/pulls/${{ matrix.pr_number }})
+          echo "ref=$(echo $PR_DATA | jq ".head.ref") >> $GITHUB_ENV"
```

Comment:
I'm not sure about the quoting of this line, but I might be getting confused as I always do about the quoting rules in yaml and bash and how they play together. If this line is sent verbatim to the shell, then it runs:

```sh
echo "ref=$(echo $PR_DATA | jq ".head.ref") >> $GITHUB_ENV"
```

Which would print, for example:

```
ref="chore/1579-remove-extending-organisation-controller" >> /path/to/github.env
```

Rather than appending the line `ref="chore/1579-remove-extending-organisation-controller"` to the file `/path/to/github.env`. Is that what's intended.

I'm also not sure if you're deliberately relying on the fact that `jq` puts `"` quotes around its output unless you use the `-r` switch, in order to make sure the appended line quotes the value of `ref`. A possibly safer version would do:

```sh
ref='$(echo $PR_DATA | jq -r ".head.ref")'
```

Since that avoids the possibility of a refname containing symbols with special meaning in bash that then get interpreted when inside `"` quotes.

Reply:
The way I think this should be quoted, given the above, is:

```
echo "ref='$(echo "$PR_DATA" | jq -r .head.ref)'" >> "$GITHUB_ENV"
```

Reply:
Ah yeah, that looks better. Too many quotes in a YAML document tend to make my eyes bleed

Reply:
An even safer way that makes sure the variable is correctly quoted in the environment file would be:

```sh
printf 'ref=%q\n' "$(echo "$PR_DATA" | jq -r .head.ref)" >> "$GITHUB_ENV"
```

Reply:
Oh nice, that's even better